From 3233a6174c6774bce2d083d94d666669da8801c0 Mon Sep 17 00:00:00 2001
From: Jayachandran B <jayachandran.b@intel.com>
Date: Thu, 27 Nov 2014 10:22:32 +0530
Subject: [PATCH 08/24] ASoC: Intel: Separate the stream maps

Separate out the stream maps for various platforms.

Issue: GMINL-4319

Change-Id: I39730767fae779f4c11643dea4cf094de679b96e
Signed-off-by: Jayachandran B <jayachandran.b@intel.com>
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 arch/x86/include/asm/intel_sst_mrfld.h     | 36 -------------
 arch/x86/include/asm/platform_byt_audio.h  | 22 +-------
 arch/x86/include/asm/platform_cht_audio.h  |  2 +-
 sound/soc/intel/board/byt_cr_dpcm_rt5640.c |  4 +-
 sound/soc/intel/board/byt_cr_dpcm_rt5651.c |  4 +-
 sound/soc/intel/board/cht_bl_dpcm_rt5672.c |  2 +-
 sound/soc/intel/pcm.c                      | 69 +------------------------
 sound/soc/intel/sst/sst_platform.c         | 82 ++++++++++--------------------
 8 files changed, 34 insertions(+), 187 deletions(-)
 delete mode 100644 arch/x86/include/asm/intel_sst_mrfld.h

diff --git a/arch/x86/include/asm/intel_sst_mrfld.h b/arch/x86/include/asm/intel_sst_mrfld.h
deleted file mode 100644
index 2afbd6c..0000000
--- a/arch/x86/include/asm/intel_sst_mrfld.h
+++ /dev/null
@@ -1,36 +0,0 @@
-/* intel_sst_mrlfd.h - Common enum of the Merrifield platform
- *
- *  Copyright (C) 2013 Intel Corp
- *  Author: Samreen Nilofer <samreen.nilofer@intel.com>
- *  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
- *
- *  This program is free software; you can redistribute it and/or modify
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation; version 2 of the License.
- *
- *  This program is distributed in the hope that it will be useful, but
- *  WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *  General Public License for more details.
- *
- * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
- */
-#ifndef _INTEL_SST_MRFLD_H
-#define _INTEL_SST_MRFLD_H
-
-enum {
-	MERR_SALTBAY_AUDIO = 0,
-	MERR_SALTBAY_COMPR,
-	MERR_SALTBAY_VOIP,
-	MERR_SALTBAY_PROBE,
-	MERR_SALTBAY_AWARE,
-	MERR_SALTBAY_VAD,
-	MERR_SALTBAY_POWER,
-};
-
-enum {
-	MERR_DPCM_AUDIO = 0,
-	MERR_DPCM_VOIP,
-};
-
-#endif
diff --git a/arch/x86/include/asm/platform_byt_audio.h b/arch/x86/include/asm/platform_byt_audio.h
index d24b6f3..d5c586a 100644
--- a/arch/x86/include/asm/platform_byt_audio.h
+++ b/arch/x86/include/asm/platform_byt_audio.h
@@ -13,30 +13,10 @@
 #define _PLATFORM_BYT_AUDIO_H_
 
 enum {
-	BYT_AUD_AIF1 = 0,
-	BYT_AUD_AIF2,
-	BYT_AUD_COMPR_DEV,
-#ifdef CONFIG_SND_SOC_COMMS_SSP
-	BYT_COMMS_BT,
-	BYT_COMMS_MODEM,
-#endif /* CONFIG_SND_SOC_COMMS_SSP */
-	BYT_AUD_PROBE_DEV,
-};
-
-enum {
-	BYT_DPCM_AUD_AIF1 = 0,
-	BYT_DPCM_DB,
-	BYT_DPCM_LL,
-	BYT_DPCM_COMPR,
+	BYT_DPCM_AUDIO = 0,
 	BYT_DPCM_VOIP,
-	BYT_DPCM_PROBE,
 };
 
-enum {
-	BYT_CR_AUD_AIF1 = 0,
-	BYT_CR_AUD_COMPR_DEV,
-	BYT_CR_COMMS_BT,
-};
 /* LPE viewpoint addresses */
 /* TODO: move to DSDT */
 #define SST_BYT_IRAM_PHY_START	0xff2c0000
diff --git a/arch/x86/include/asm/platform_cht_audio.h b/arch/x86/include/asm/platform_cht_audio.h
index c40adcc..7b7312c 100644
--- a/arch/x86/include/asm/platform_cht_audio.h
+++ b/arch/x86/include/asm/platform_cht_audio.h
@@ -13,7 +13,7 @@
 #define _PLATFORM_CHT_AUDIO_H_
 
 enum {
-	CHT_DPCM_AUD_AIF1 = 0,
+	CHT_DPCM_AUDIO = 0,
 	CHT_DPCM_VOIP,
 };
 
diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
index 1c4ce4f..f6375bf 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
@@ -793,7 +793,7 @@ static const struct snd_soc_pcm_stream byt_dai_params = {
 };
 
 static struct snd_soc_dai_link byt_dailink[] = {
-	{
+	[BYT_DPCM_AUDIO] = {
 		.name = "Baytrail Audio Port",
 		.stream_name = "Baytrail Audio",
 		.cpu_dai_name = "Headset-cpu-dai",
@@ -805,7 +805,7 @@ static struct snd_soc_dai_link byt_dailink[] = {
 		.dynamic = 1,
 		.ops = &byt_aif1_ops,
 	},
-	{
+	[BYT_DPCM_VOIP] = {
 		.name = "Baytrail VOIP Port",
 		.stream_name = "Baytrail Voip",
 		.cpu_dai_name = "Voip-cpu-dai",
diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5651.c b/sound/soc/intel/board/byt_cr_dpcm_rt5651.c
index 345bdc4..456417e 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5651.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5651.c
@@ -772,7 +772,7 @@ static struct snd_soc_ops byt_8k_16k_ops = {
 };
 
 static struct snd_soc_dai_link byt_dailink[] = {
-	{
+	[BYT_DPCM_AUDIO] = {
 		.name = "Baytrail Audio Port",
 		.stream_name = "Baytrail Audio",
 		.cpu_dai_name = "Headset-cpu-dai",
@@ -784,7 +784,7 @@ static struct snd_soc_dai_link byt_dailink[] = {
 		.dynamic = 1,
 		.ops = &byt_aif1_ops,
 	},
-	{
+	[BYT_DPCM_VOIP] = {
 		.name = "Baytrail VOIP Port",
 		.stream_name = "Baytrail Voip",
 		.cpu_dai_name = "Voip-cpu-dai",
diff --git a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
index 611a9f1..30209aa 100644
--- a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
+++ b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
@@ -749,7 +749,7 @@ static struct snd_soc_ops cht_be_ssp2_ops = {
 
 
 static struct snd_soc_dai_link cht_dailink[] = {
-	[CHT_DPCM_AUD_AIF1] = {
+	[CHT_DPCM_AUDIO] = {
 		.name = "Cherrytrail Audio Port",
 		.stream_name = "Cherrytrail Audio",
 		.cpu_dai_name = "Headset-cpu-dai",
diff --git a/sound/soc/intel/pcm.c b/sound/soc/intel/pcm.c
index 4c38901..c80159b 100644
--- a/sound/soc/intel/pcm.c
+++ b/sound/soc/intel/pcm.c
@@ -33,7 +33,6 @@
 #include <sound/soc.h>
 #include <sound/intel_sst_ioctl.h>
 #include <asm/platform_sst_audio.h>
-#include <asm/intel_sst_mrfld.h>
 #include <asm/intel-mid.h>
 #include "platform_ipc_v2.h"
 #include "sst_platform.h"
@@ -81,15 +80,6 @@ static struct snd_pcm_hardware sst_platform_pcm_hw = {
 	.fifo_size = SST_FIFO_SIZE,
 };
 
-static struct sst_dev_stream_map dpcm_strm_map[] = {
-	{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}, /* Reserved, not in use */
-	{MERR_DPCM_AUDIO, 0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_MEDIA1_IN, SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
-	{MERR_DPCM_VOIP,  0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_VOIP_IN, SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
-	{MERR_DPCM_AUDIO, 0, SNDRV_PCM_STREAM_CAPTURE, PIPE_PCM1_OUT, SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
-	{MERR_DPCM_VOIP,  0, SNDRV_PCM_STREAM_CAPTURE, PIPE_VOIP_OUT, SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
-	/* stream ID 25 used by Aware, but no device exposed to userspace */
-};
-
 static int sst_platform_ihf_set_tdm_slot(struct snd_soc_dai *dai,
 			unsigned int tx_mask, unsigned int rx_mask,
 			int slots, int slot_width) {
@@ -182,16 +172,11 @@ static void sst_fill_pcm_params(struct snd_pcm_substream *substream,
 
 }
 
-#define ASSIGN_PIPE_ID(periodtime, lowlatency, deepbuffer) \
-	((periodtime) <= (lowlatency) ? PIPE_LOW_PCM0_IN : \
-	((periodtime) >= (deepbuffer) ? PIPE_MEDIA3_IN : PIPE_MEDIA1_IN))
-
 static int sst_get_stream_mapping(int dev, int sdev, int dir,
 	struct sst_dev_stream_map *map, int size, u8 pipe_id,
 	const struct sst_lowlatency_deepbuff *ll_db)
 {
 	int index;
-	unsigned long pt = 0, ll = 0, db = 0;
 
 	if (map == NULL)
 		return -EINVAL;
@@ -204,32 +189,8 @@ static int sst_get_stream_mapping(int dev, int sdev, int dir,
 		    (map[index].subdev_num == sdev) &&
 		    (map[index].direction == dir)) {
 			/* device id for the probe is assigned dynamically */
-			if (map[index].status == SST_DEV_MAP_IN_USE) {
-				return index;
-			} else if (map[index].status == SST_DEV_MAP_FREE) {
-				map[index].status = SST_DEV_MAP_IN_USE;
-
-				if (map[index].dev_num == MERR_SALTBAY_PROBE) {
-					map[index].device_id = pipe_id;
-
-				} else if (map[index].dev_num == MERR_SALTBAY_AUDIO) {
-					if (!ll_db->low_latency || !ll_db->deep_buffer)
-						return -EINVAL;
-
-					pt = ll_db->period_time;
-					ll = *(ll_db->low_latency);
-					db = *(ll_db->deep_buffer);
-
-					pr_debug("PT %lu LL %lu DB %lu\n", pt, ll, db);
-
-					map[index].device_id = ASSIGN_PIPE_ID(pt,
-								ll, db);
-				}
-				pr_debug("%s: pipe_id 0%x index %d", __func__,
-						map[index].device_id, index);
-
+			if (map[index].status == SST_DEV_MAP_IN_USE)
 				return index;
-			}
 		}
 	}
 	return 0;
@@ -439,33 +400,11 @@ out_ops:
 	return ret_val;
 }
 
-static void sst_free_stream_in_use(struct sst_dev_stream_map *map, int str_id)
-{
-	if (dpcm_enable == 1)
-		return;
-
-	if ((map[str_id].dev_num == MERR_SALTBAY_AUDIO) ||
-			(map[str_id].dev_num == MERR_SALTBAY_PROBE)) {
-		/* Do nothing in capture for audio device */
-		if ((map[str_id].dev_num == MERR_SALTBAY_AUDIO) &&
-				(map[str_id].direction == SNDRV_PCM_STREAM_CAPTURE))
-			return;
-		if ((map[str_id].task_id == SST_TASK_ID_MEDIA) &&
-				(map[str_id].status == SST_DEV_MAP_IN_USE)) {
-			pr_debug("str_id %d device_id 0x%x\n", str_id, map[str_id].device_id);
-			map[str_id].status = SST_DEV_MAP_FREE;
-			map[str_id].device_id = PIPE_RSVD;
-		}
-	}
-	return;
-}
-
 static void sst_media_close(struct snd_pcm_substream *substream,
 		struct snd_soc_dai *dai)
 {
 	struct sst_runtime_stream *stream;
 	int ret_val = 0, str_id;
-	struct sst_data *ctx = snd_soc_platform_get_drvdata(dai->platform);
 
 	stream = substream->runtime->private_data;
 	if (strstr(dai->name, "Power-cpu-dai"))
@@ -474,7 +413,6 @@ static void sst_media_close(struct snd_pcm_substream *substream,
 	str_id = stream->stream_info.str_id;
 	if (str_id)
 		ret_val = stream->ops->close(str_id);
-	sst_free_stream_in_use(ctx->pdata->pdev_strm_map, str_id);
 	module_put(sst_dsp->dev->driver->owner);
 	kfree(stream);
 	pr_debug("%s: %d\n", __func__, ret_val);
@@ -963,11 +901,6 @@ static int sst_platform_probe(struct platform_device *pdev)
 		return -ENOMEM;
 	}
 
-	if (dpcm_enable == 1) {
-		pr_debug("dpcm enabled; overriding stream map\n");
-		pdata->pdev_strm_map = dpcm_strm_map;
-		pdata->strm_map_size = ARRAY_SIZE(dpcm_strm_map);
-	}
 	sst_pdev = &pdev->dev;
 	sst->pdata = pdata;
 	mutex_init(&sst->lock);
diff --git a/sound/soc/intel/sst/sst_platform.c b/sound/soc/intel/sst/sst_platform.c
index ea93572..6759aee 100644
--- a/sound/soc/intel/sst/sst_platform.c
+++ b/sound/soc/intel/sst/sst_platform.c
@@ -18,64 +18,34 @@
 #include <linux/platform_device.h>
 #include <asm/platform_sst_audio.h>
 #include <asm/intel-mid.h>
-#include <asm/intel_sst_mrfld.h>
 #include <asm/platform_byt_audio.h>
+#include <asm/platform_cht_audio.h>
 #include <sound/asound.h>
 #include "sst.h"
 
 static struct sst_platform_data sst_platform_pdata;
-static struct sst_dev_stream_map mrfld_strm_map[] = {
+static struct sst_dev_stream_map dpcm_strm_map_byt[] = {
 	{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}, /* Reserved, not in use */
-	{MERR_SALTBAY_AUDIO, 0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_AUDIO, 1, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_AUDIO, 2, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_COMPR, 0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_MEDIA0_IN,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
-	{MERR_SALTBAY_VOIP, 0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_VOIP_IN,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
-	{MERR_SALTBAY_AUDIO, 0, SNDRV_PCM_STREAM_CAPTURE, PIPE_PCM1_OUT,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
-	{MERR_SALTBAY_VOIP, 0, SNDRV_PCM_STREAM_CAPTURE, PIPE_VOIP_OUT,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
-	{MERR_SALTBAY_PROBE, 0, SNDRV_PCM_STREAM_CAPTURE, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 1, SNDRV_PCM_STREAM_CAPTURE, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 2, SNDRV_PCM_STREAM_CAPTURE, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 3, SNDRV_PCM_STREAM_CAPTURE, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 4, SNDRV_PCM_STREAM_CAPTURE, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 5, SNDRV_PCM_STREAM_CAPTURE, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 6, SNDRV_PCM_STREAM_CAPTURE, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 7, SNDRV_PCM_STREAM_CAPTURE, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 1, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 2, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 3, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 4, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 5, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 6, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_PROBE, 7, SNDRV_PCM_STREAM_PLAYBACK, PIPE_RSVD,
-	SST_TASK_ID_MEDIA, SST_DEV_MAP_FREE},
-	{MERR_SALTBAY_AWARE, 0, SNDRV_PCM_STREAM_CAPTURE, PIPE_AWARE_OUT,
-	SST_TASK_ID_AWARE, SST_DEV_MAP_IN_USE},
-	{MERR_SALTBAY_VAD, 0, SNDRV_PCM_STREAM_CAPTURE, PIPE_VAD_OUT,
-	SST_TASK_ID_AWARE, SST_DEV_MAP_IN_USE},
+	{BYT_DPCM_AUDIO, 0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_MEDIA1_IN,
+					SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
+	{BYT_DPCM_VOIP,  0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_VOIP_IN,
+					SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
+	{BYT_DPCM_AUDIO, 0, SNDRV_PCM_STREAM_CAPTURE, PIPE_PCM1_OUT,
+					SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
+	{BYT_DPCM_VOIP,  0, SNDRV_PCM_STREAM_CAPTURE, PIPE_VOIP_OUT,
+					SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
+};
+
+static struct sst_dev_stream_map dpcm_strm_map_cht[] = {
+	{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}, /* Reserved, not in use */
+	{CHT_DPCM_AUDIO, 0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_MEDIA1_IN,
+					SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
+	{CHT_DPCM_VOIP,  0, SNDRV_PCM_STREAM_PLAYBACK, PIPE_VOIP_IN,
+					SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
+	{CHT_DPCM_AUDIO, 0, SNDRV_PCM_STREAM_CAPTURE, PIPE_PCM1_OUT,
+					SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
+	{CHT_DPCM_VOIP,  0, SNDRV_PCM_STREAM_CAPTURE, PIPE_VOIP_OUT,
+					SST_TASK_ID_MEDIA, SST_DEV_MAP_IN_USE},
 };
 
 static const int sst_ssp_mux_shift[SST_NUM_SSPS] = {
@@ -292,8 +262,8 @@ sst_ssp_configs_cht_cr[SST_NUM_SSPS][SST_MAX_SSP_MUX][SST_MAX_SSP_DOMAINS] = {
 
 static void set_cht_platform_config(void)
 {
-	sst_platform_pdata.pdev_strm_map = mrfld_strm_map;
-	sst_platform_pdata.strm_map_size = ARRAY_SIZE(mrfld_strm_map);
+	sst_platform_pdata.pdev_strm_map = dpcm_strm_map_cht;
+	sst_platform_pdata.strm_map_size = ARRAY_SIZE(dpcm_strm_map_cht);
 	sst_platform_pdata.dfw_enable = 1;
 	memcpy(sst_platform_pdata.ssp_config, sst_ssp_configs_mrfld, sizeof(sst_ssp_configs_mrfld));
 	memcpy(sst_platform_pdata.mux_shift, sst_ssp_mux_shift, sizeof(sst_ssp_mux_shift));
@@ -303,8 +273,8 @@ static void set_cht_platform_config(void)
 
 static void set_cht_cr_platform_config(void)
 {
-	sst_platform_pdata.pdev_strm_map = mrfld_strm_map;
-	sst_platform_pdata.strm_map_size = ARRAY_SIZE(mrfld_strm_map);
+	sst_platform_pdata.pdev_strm_map = dpcm_strm_map_byt;
+	sst_platform_pdata.strm_map_size = ARRAY_SIZE(dpcm_strm_map_byt);
 	sst_platform_pdata.dfw_enable = 0;
 	memcpy(sst_platform_pdata.ssp_config, sst_ssp_configs_cht_cr, sizeof(sst_ssp_configs_cht_cr));
 	memcpy(sst_platform_pdata.mux_shift, sst_ssp_mux_shift, sizeof(sst_ssp_mux_shift));
-- 
1.9.1

